src/node/rpc/node_frontend.h

```diff
@@ -1377,34 +1392,41 @@ namespace ccf
             "Configurations: sufficient number of ORCs, updating nodes in "
             "configuration #{}",
             in.reconfiguration_id);
-          auto ncfgs = args.tx.ro(network.network_configurations);
+          auto ncfgs = args.tx.rw(network.network_configurations);
           auto nodes = args.tx.rw(network.nodes);
           auto nc = ncfgs->get(in.reconfiguration_id);
-          for (auto nid : nc->nodes)
+
+          if (!nc.has_value())
           {
-            auto node_info = nodes->get(nid);
-            if (!node_info.has_value())
-            {
-              return make_error(
-                HTTP_STATUS_INTERNAL_SERVER_ERROR,
-                ccf::errors::InternalError,
-                fmt::format("Missing node information for {}", nid));
-            }
+            return make_error(
+              HTTP_STATUS_BAD_REQUEST,
+              ccf::errors::ResharingAlreadyCompleted,
```

Comment:
This isn't about resharing, is it?

Reply:
Nope, good catch, that's a copy&paste gone wrong :)